src/dexnav.c

```diff
@@ -0,0 +1,2715 @@
+            task->func = Task_DexNavExitAndSearch;
+        }
+    }
+}
+
+/////////////////////////
+//// HIDDEN POKEMON /////
+/////////////////////////
+bool8 TryFindHiddenPokemon(void)
+{
+    u16 *stepPtr = GetVarPointer(VAR_DEXNAV_STEP_COUNTER);
+    
+    if (!FlagGet(FLAG_SYS_DETECTOR_MODE) || FlagGet(FLAG_SYS_DEXNAV_SEARCH) || GetFlashLevel() > 0)
+    {
+        (*stepPtr) = 0;
+        return FALSE;
+    }
+    
+    (*stepPtr)++;
+    (*stepPtr) %= HIDDEN_MON_STEP_COUNT;
+    if ((*stepPtr) == 0 && (Random() % 100 < HIDDEN_MON_SEARCH_RATE))
+    {
+        // hidden pokemon
+        u16 headerId = GetCurrentMapWildMonHeaderId();
+        u8 index;
+        u16 species;
+        u8 environment;
+        u8 taskId;
+        const struct WildPokemonInfo* hiddenMonsInfo = gWildMonHeaders[headerId].hiddenMonsInfo;
+        bool8 isHiddenMon = FALSE;
+        
+        //while you can still technically find hidden pokemon if there are not hidden-only pokemon on a map,
+        // this prevents any potential lagging on maps you dont want hidden pokemon to appear on
+        if (hiddenMonsInfo == NULL)
+            return FALSE;
+        
+        //encounter rate signifies surfing (1) or land mons (0)!
+        // again, for simplicity
+        switch (hiddenMonsInfo->encounterRate)
+        {
+        case 0: //land
+            // there are surely better ways to do this, but this allows greatest flexibility
+            if (Random() % 100 < HIDDEN_MON_PROBABILTY)
+            {
+                index = ChooseHiddenMonIndex();
+                if (index == 0xFF)
+                    return FALSE;//no hidden info
+                species = hiddenMonsInfo->wildPokemon[index].species;
+                isHiddenMon = TRUE;
+                environment = ENCOUNTER_TYPE_HIDDEN;
+            }
+            else
+            {
+                species = gWildMonHeaders[headerId].landMonsInfo->wildPokemon[ChooseWildMonIndex_Land()].species;
+                environment = ENCOUNTER_TYPE_LAND;
+            }
+            break;
+        case 1: //water
+            if (TestPlayerAvatarFlags(PLAYER_AVATAR_FLAG_SURFING))
+            {
+                if (Random() % 100 < HIDDEN_MON_PROBABILTY)
+                {
+                    index = ChooseHiddenMonIndex();
+                    if (index == 0xFF)
+                        return FALSE;//no hidden info
+                    species = hiddenMonsInfo->wildPokemon[index].species;
+                    isHiddenMon = TRUE;
+                    environment = ENCOUNTER_TYPE_HIDDEN;
+                }
+                else
+                {
+                    species = gWildMonHeaders[headerId].waterMonsInfo->wildPokemon[ChooseWildMonIndex_WaterRock()].species;
+                    environment = ENCOUNTER_TYPE_WATER;
+
+                }
+            }
+            else
+            {
+                return FALSE;   //not surfing -> cant find hidden water mons
```

Comment:
comments aren't always inlined the same way either, sometimes they're directly after, sometimes there's a space between, and sometimes it looks like there's a tab